images/rabbitmq-default-user-credential-updater/tests/main.tf

```diff
@@ -0,0 +1,116 @@
+terraform {
+  required_providers {
+    oci       = { source = "chainguard-dev/oci" }
+    imagetest = { source = "chainguard-dev/imagetest" }
+  }
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+locals { parsed = provider::oci::parse(var.digest) }
+
+data "imagetest_inventory" "this" {}
+
+resource "imagetest_harness_k3s" "this" {
+  name      = "rabbitmq-default-user-credential-updater"
+  inventory = data.imagetest_inventory.this
+
+  sandbox = {
+    mounts = [
+      {
+        source      = path.module
+        destination = "/tests"
+      }
+    ]
+  }
+}
+
+locals {
+  namespace = "rabbitmq-system"
+}
+
+module "helm-vault" {
+  source = "../../../tflib/imagetest/helm"
+
+  repo      = "https://helm.releases.hashicorp.com"
+  chart     = "vault"
+  namespace = local.namespace
+  name      = "vault"
+
+  values = {
+    server = {
+      dev      = { enabled = true }
+      logLevel = "debug"
+    }
+    injector = {
+      logLevel = "debug"
+    }
+  }
+}
+
+resource "imagetest_feature" "vault" {
+  harness     = imagetest_harness_k3s.this
+  name        = "Vault"
+  description = "Integration test with RabbitMQ and Hashicorp Vault."
+
+  steps = [
+    {
+      name = "Create required namespace"
+      cmd  = <<EOF
+        kubectl create ns ${local.namespace}
+      EOF
+    },
+    {
+      name = "Helm deploy Vault"
+      cmd  = module.helm-vault.install_cmd
+    },
+    {
+      name = "Wait for deployment to finish, then run setup.sh script"
+      cmd  = <<EOF
+        sleep 5
+        kubectl wait --for=condition=Ready pod/vault-0 -n ${local.namespace}
+        chmod +x /tests/setup.sh
+        RABBITMQ_NAMESPACE=${local.namespace} /tests/setup.sh
+      EOF
+    },
+    {
+      name = "Depoly rabbit mq with custom default-user-credential-updater image"
+      cmd  = <<EOF
+        apk add kustomize
+cat > kustomization.yaml <<kust
+resources:
+- https://github.com/rabbitmq/cluster-operator/releases/latest/download/cluster-operator.yml
+patches:
+- patch: |-
+    - op: add
+      path: "/spec/template/spec/containers/0/env/-"
+      value: 
+        name: DEFAULT_USER_UPDATER_IMAGE
+        value: "${local.parsed.registry_repo}:${local.parsed.pseudo_tag}"
```

Comment:
```suggestion
        value: "${var.digest}"
```

Reply:
Turns out this is all we need to get the image from the registry in the test harness.

Reply:
This change caused the tests not to pass so I reverted this change back to the original.